pkg/util/finalizer.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2022 The Kruise Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type FinalizerOpType string
+
+const (
+	AddFinalizerOpType    FinalizerOpType = "Add"
+	RemoveFinalizerOpType FinalizerOpType = "Remove"
+)
+
+func UpdateFinalizer(c client.Client, object client.Object, op FinalizerOpType, finalizer string) error {
```

Comment:
We don't have to define `UpdateFinalizer` function, just use Patch in the two places that want to add or remove finalizer.